esrally/driver/driver.py

```diff
@@ -516,14 +516,14 @@ def receiveMsg_PrepareTrack(self, msg, sender):
             self.processors.put(processor)
         self._seed_tasks(self.processors.get())
         self.send_to_children_and_transition(
-            self, StartTaskLoop(self.track.name, self.cfg), self.Status.INITIALIZING, self.Status.PROCESSOR_RUNNING
+            self, StartTaskLoop(self.track.name, self.cfg), self.Status.INITIALIZING, self.Status.PROCESSOR_RUNNING  # type: ignore[arg-type]  # pylint: disable=C0301  # TODO remove this ignore when introducing type hints
```

Comment:
Reading the code, it seems to me that `self.cfg` will probably never be None when passed to `StartTaskLoop()`. How about prepending `assert self.cfg is not None` instead of making the argument to `Optional[config.Config]`?

Reply:
Agreed, that's better :)

Reply:
Removed `# type: ignore[arg-type]` by `assert self.cfg is not None` accordingly.